Blockzilla/Onboarding/OnboardingEventsHandler.swift

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Combine
+
+class OnboardingEventsHandler {
+    
+    enum Action {
+        case applicationDidLaunch
+        case onboardingDidDismiss
+        case startBrowsing
+        case resetBrowser
+        case showTrackingProtection
+    }
+    
+    @Published var shouldPresentOnboarding: Bool = false
+    @Published var shouldPresentShieldToolTip: Bool = false
+    @Published var shouldPresentTrashToolTip: Bool = false
+    @Published var shouldPresentMenuToolTip: Bool = false
+    @Published var shouldPresentTrackingProtectionToolTip: Bool = false
+    
+    private var visitedURLcounter = 0
+    private var menuToolTipDidAppear = false
+    private var trackingProtectionToolTipDidAppear = false
+    
```

Comment:
Too many boolean states. I think we should refine this logic. For now we can leave it.

Reply:
I will leave it for now as you suggested, because when we will start testing with the proper UI and also with the flag for showing old onboarding vs new onboarding, I think we should add some other checks in the logic. Thank you